micro-rdk/src/common/ota.rs

```diff
@@ -189,20 +192,40 @@ impl OtaService {
             );
             uri = hyper::Uri::from_parts(parts).map_err(|e| OtaError::Other(e.to_string()))?;
         };
-        let io = self
-            .connector
-            .connect_to(&uri)
-            .map_err(|e| OtaError::Other(e.to_string()))?
-            .await
-            .map_err(|e| OtaError::Other(e.to_string()))?;
 
-        let (mut sender, conn) = {
-            http2::Builder::new(self.exec.clone())
-                .max_frame_size(16_384) // lowest configurable value
-                .timer(H2Timer)
-                .handshake(io)
-                .await
-                .map_err(|e| OtaError::Other(e.to_string()))?
+        let (mut sender, conn) = loop {
+            match self.connector.connect_to(&uri) {
+                Ok(connection) => {
+                    match connection.await {
+                        Ok(io) => {
+                            match http2::Builder::new(self.exec.clone())
+                                .max_frame_size(16_384) // lowest configurable
+                                .timer(H2Timer)
+                                .handshake(io)
+                                .await
+                            {
+                                Ok(pair) => break pair,
+                                Err(e) => {
+                                    log::error!("failed to build http request: {}", e);
+                                }
+                            }
+                        }
+                        Err(e) => {
+                            log::error!("failed to create tcp stream: {}", e);
+                        }
+                    }
+                }
+                Err(e) => {
+                    log::error!("failed to create http connection: {}", e);
+                }
+            };
+
+            log::debug!(
+                "retry establishing http connection to {} in {} seconds",
+                &uri,
```

Comment:
@acmorrow added the uri that was tried here and modified the other logs slightly. As the other errors will also hit this log I don't think it's necessary to add the uri to them as well, especially since the uri's with embedded permissions tend to be very lengthy, though can add it if you want.

Other than that, I'm sure what more I can say above that would provide better context than surfacing the underlying error.